src/MartinGeorgiev/Doctrine/ORM/Query/AST/Functions/Tsrange.php

```diff
@@ -12,12 +12,19 @@
  *
  * @author Martin Georgiev <martin.georgiev@gmail.com>
  */
-class Tsrange extends BaseFunction
+class Tsrange extends BaseRangeFunction
 {
-    protected function customizeFunction(): void
+    protected function getFunctionName(): string
     {
-        $this->setFunctionPrototype('tsrange(%s, %s)');
-        $this->addNodeMapping('StringPrimary');
-        $this->addNodeMapping('StringPrimary');
+        return 'tsrange';
+    }
+
+    protected function getNodeMappingPattern(): array
+    {
+        return [
+            'StringPrimary',
+            'StringPrimary',
+            'StringPrimary',
```

Comment:
@coderabbitai - does this look alright to you? Analyse the usage of `getNodeMappingPattern` and cross-check with Postgresql's function signature. You may need to revise similar changes in the current PR.

Reply:
Are you sure? Explain why RegexpCount and RegexpInstr look different then.

Reply:
Thanks bunny <3 :D

Reply:
Sorry, I may have interrupted the AI flow.

Reply:
@coderabbitai come back

Reply:
@coderabbitai Then why do you think 
```php
protected function getNodeMappingPattern(): array
{
    return [
        'StringPrimary',
        'StringPrimary',
        'StringPrimary',
    ];
}
```
is fine? The three function signature overloads with the exact same pattern make no sense to me (`'StringPrimary',`).

Reply:
@coderabbitai, you seem to be wrong. When there is a single string (like in Tsrange), it will apply to each element as the same parameter type. Explore very carefully the implementation of BaseVariadicFunction and how it is used differently for some other existing classes.

Reply:
You got most of it, but if the code changes to 
```php
return [
    'StringPrimary,StringPrimary,StringPrimary',
];
```
you will introduce another bug. Can you figure out what that will be?

Reply:
@coderabbitai - where did you go? See the comments above.

Reply:
No, your response regressed. Why didn't you think of simply having, in this instance: 
```
protected function getNodeMappingPattern(): array
{
    return ['StringPrimary'];
}
```
Learn this.

Reply:
@coderabbitai look up

Reply:
@coderabbitai Now revise your approval of the PR and request changes where they are relevant!